scripts/pylib/twister/twisterlib.py

```diff
@@ -972,6 +972,8 @@ def handle(self):
                 self.instance.reason = "Failed"
         else:
             self.instance.execution_time = handler_time
+            self.instance.status = "failed"
+            self.instance.reason = "No Console Output"
```

Comment:
if code goes here, the harness.state is not set(None), which only happens no console output, so that all the checking is skiped, with this change it log will show 'No Console Output', and you will see timeout in xml report.

Reply:
But if there are some testcases already passed in this testsuite, take following example:

```
DEBUG   - DEVICE: Running TESTSUITE posix_apis
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_pthread_execution
DEBUG   - DEVICE: Thread 0 starting with scheduling policy 0 & priority 15
DEBUG   - DEVICE: Thread 1 starting with scheduling policy 0 & priority 15
DEBUG   - DEVICE: Thread 2 starting with scheduling policy 0 & priority 15
DEBUG   - DEVICE: Bounce test OK
DEBUG   - DEVICE: Barrier test OK
DEBUG   - DEVICE: PASS - test_posix_pthread_execution in 0.521 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_pthread_error_condition
DEBUG   - DEVICE: PASS - test_posix_pthread_error_condition in 0.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_pthread_termination
DEBUG   - DEVICE: Thread 0 starting with a priority of 4
DEBUG   - DEVICE: Cancelling thread 0
DEBUG   - DEVICE: Thread 1 starting with a priority of 3
DEBUG   - DEVICE: Cancelling thread 1
DEBUG   - DEVICE: Thread 1 could not be cancelled
DEBUG   - DEVICE: Thread 2 starting with a priority of 2
DEBUG   - DEVICE: Cancelling thread 2
DEBUG   - DEVICE: Thread 3 starting with a priority of 1
DEBUG   - DEVICE: Cancelling thread 3
DEBUG   - DEVICE: Thread 3 could not be cancelled
DEBUG   - DEVICE: PASS - test_posix_pthread_termination in 1.16 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_multiple_threads_single_key
DEBUG   - DEVICE:
DEBUG   - DEVICE: Different threads set different values to same key:
DEBUG   - DEVICE: thread 0: set value = 6214384 and retrieved value = 6214384
DEBUG   - DEVICE: thread 1: set value = 6214832 and retrieved value = 6214832
DEBUG   - DEVICE:
DEBUG   - DEVICE: PASS - test_posix_multiple_threads_single_key in 0.18 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_single_thread_multiple_keys
DEBUG   - DEVICE:
DEBUG   - DEVICE: Single thread associates its value with different keys:
DEBUG   - DEVICE: key 0: set value = 6215232 and retrieved value = 6215232
DEBUG   - DEVICE: key 1: set value = 6215232 and retrieved value = 6215232
DEBUG   - DEVICE:
DEBUG   - DEVICE: PASS - test_posix_single_thread_multiple_keys in 0.18 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_thread_attr_stacksize
DEBUG   - DEVICE: PASS - test_posix_thread_attr_stacksize in 0.1 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_clock
DEBUG   - DEVICE: POSIX clock APIs
DEBUG   - DEVICE: POSIX clock APIs test done
DEBUG   - DEVICE: PASS - test_posix_clock in 2.5 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_semaphore
DEBUG   - DEVICE: PASS - test_posix_semaphore in 2.305 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_normal_mutex
DEBUG   - DEVICE: mutex lock is taken
DEBUG   - DEVICE: PASS - test_posix_normal_mutex in 0.203 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_recursive_mutex
DEBUG   - DEVICE: recursive mutex lock is taken
DEBUG   - DEVICE: PASS - test_posix_recursive_mutex in 0.4 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_mqueue
DEBUG   - DEVICE: PASS - test_posix_mqueue in 0.11 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_realtime
DEBUG   - DEVICE: PASS - test_posix_realtime in 1.802 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_timer
DEBUG   - DEVICE: POSIX timer test
DEBUG   - DEVICE: Timer fires every 0 secs and  100000000 nsecs
DEBUG   - DEVICE: Time remaining to fire 0 secs and  899000000 nsecs
DEBUG   - DEVICE: Handler Signal value :20 for 1 times
DEBUG   - DEVICE: Handler Signal value :20 for 2 times
DEBUG   - DEVICE: Handler Signal value :20 for 3 times
DEBUG   - DEVICE: Handler Signal value :20 for 4 times
DEBUG   - DEVICE: Handler Signal value :20 for 5 times
DEBUG   - DEVICE: Handler Signal value :20 for 6 times
DEBUG   - DEVICE: Handler Signal value :20 for 7 times
DEBUG   - DEVICE: Handler Signal value :20 for 8 times
DEBUG   - DEVICE: Handler Signal value :20 for 9 times
DEBUG   - DEVICE: Handler Signal value :20 for 10 times
DEBUG   - DEVICE: Handler Signal value :20 for 11 times
DEBUG   - DEVICE: Handler Signal value :20 for 12 times
DEBUG   - DEVICE: PASS - test_posix_timer in 2.112 seconds
DEBUG   - DEVICE: ===================================================================
DEBUG   - DEVICE: START - test_posix_rw_lock
DEBUG   - DEVICE:
DEBUG   - DEVICE: main acquire WR lock and 3 threads acquire RD lock
```

In this case, when testcases hang up in test_posix_rw_lock( ), it has console output but judge as "No console output".
This also fit to other testcases which hanging up at running.

Just a suggestion, not blocked your PR, I think your fix is good.

Reply:
But it is another 'no console output'. I get your point, it may confuse. but set to failed also looks confused

Reply:
maybe "Timeout"?

Reply:
I prefer "Timeout", because we already defined a "Timeout" reason before and it did handle this situation.

Reply:
Timeout is better I think.

Reply:
Anyway, I think this PR should be merged.